qemu/tests/balloon_service.py

```diff
@@ -0,0 +1,114 @@
+import logging
+import time
+
+from virttest import utils_test
+from virttest import qemu_monitor
+from virttest import error_context
+from avocado.core import exceptions
+from autotest.client.shared import utils
+from autotest.client.shared import error
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Balloon service test for windows guest.
+    1) boot a windows guest with balloon device.
+    2) enable and check driver verifier in guest.
+    3) install balloon service in guest.
+    4) enable polling in qmp.
+    5) evict and enlarge balloon.
+    6) get polling value in qmp.
+    7) uninstall balloon service and clear driver verifier.
+    """
+
+    def balloon_service(session):
+        """
+        Check / Install balloon service.
+
+        :param session: VM session.
+        """
+        status_balloon_service = params.get("status_balloon_service")
+        output = session.cmd_output(status_balloon_service)
+        if "RUNNING" in output:
+            logging.debug("Balloon Serice has already been installed.")
+            return
+        else:
+            try:
+                install_balloon_service = params.get("install_balloon_service")
+                session.cmd(install_balloon_service)
+            except Exception:
+                raise exceptions.TestError("Install balloon service failed !")
+
+    timeout = int(params.get("login_timeout", 360))
+    driver = params.get("driver_name", "balloon")
+
+    error_context.context("Boot guest with balloon device", logging.info)
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_login(timeout=timeout)
+
+    try:
+        error_context.context("Enable %s driver verifier in guest" % driver,
+                              logging.info)
+        if params.get("need_enable_verifier", "yes") == "yes":
+            session = utils_test.qemu.setup_win_driver_verifier(session, driver,
+                                                                vm, timeout)
+
+        error_context.context("Install balloon service in guest", logging.info)
+        balloon_service(session)
```

Comment:
from step info, it looks you want to install balloon service in guest,  but I can't  understand it from your function name